redwood/src/keys.rs

```diff
@@ -0,0 +1,55 @@
+use crate::{Error, Result};
+use sequoia_openpgp::cert::prelude::ValidErasedKeyAmalgamation;
+use sequoia_openpgp::packet::key::{PublicParts, SecretParts, UnspecifiedRole};
+use sequoia_openpgp::packet::Key;
+use sequoia_openpgp::policy::Policy;
+use sequoia_openpgp::Cert;
+
+/// We want to use the same iterators on public and secret keys but it's not
+/// really possible to do it in a function because of type differences so we
+/// use a macro instead.
+macro_rules! filter_keys {
```

Comment:
I spent a while trying to write a generic function that applied these filters over public key parts and secret key parts (distinct types) and 1) it was super complicated and 2) I couldn't actually get it to work/compile.

So I went with a macro, but I'm also okay saying macros are too magical for our collective current level of Rust and to just keep duplicating for now.

Reply:
(I vote leave it in - it is clear/readable, basically by the book, and less error-prone than manual duplication of the same logic in different places)